### PR TITLE
Remove unnecessary parens that trip errorprone in some configurations

### PR DESCRIPTION
**Pull request by kennknowles**
Remove extraneous parens. I hit this via errorprone warnings failing the build due to -Werror.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [n/a] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

R: @swegner

### CODE REVIEW

**Comment by kennknowles** [comment]
All of these fixes come from running the gradle command to "check" the SQL module, from the IntelliJ sidebar.

As far as equals goes, it seems errorprone has decided subtyping substitutability is more important than symmetry. I don't think that is an informed view, so probably better to suppress than to satisfy.

**Comment by kennknowles** [comment]
From a quick scan, I didn't find where we pin the version of errorprone. We really ought to, since it will randomly break all builds when they make a new failure. I thought we did, so maybe I just didn't find it. But clearly the checks I'm hitting haven't been hit & fixed before.

**Comment by mxm** [comment]
Seems like the `gradle-errorprone-plugin` picks a dynamic version of `errorprone` when none is in the classpath. I could only find `com.google.errorprone:error_prone_annotations:2.0.15`.

**Comment by mxm** [comment]
Tried adding `errorprone("com.google.errorprone:error_prone_core:2.1.2")` to the Java dependencies as described in https://github.com/tbroyer/gradle-errorprone-plugin/blob/master/README.md but didn't change anything.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java [hunk 1]
```diff
@@ -1541,7 +1541,7 @@ public int compare(Method o1, Method o2) {
           } else {
             try {
               Class<?> runnerClass = Class.forName(runner, true, ReflectHelpers.findClassLoader());
-              if (!(PipelineRunner.class.isAssignableFrom(runnerClass))) {
+              if (!PipelineRunner.class.isAssignableFrom(runnerClass)) {
```
[the remark this answers is not in the document]
Comment (mxm) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java]: Seems like IntelliJ has polluted my regular Gradle environment. I'm only getting this after I tried builiding with IntelliJ.

**Comment by tweise** [comment]
Should we turn off `-Werror` until the issue is fixed?

**Comment by mxm** [comment]
Disabling `errorprone` would suffice. It's a curious problem, there hasn't been an `errorprone` release in a while: https://mvnrepository.com/artifact/com.google.errorprone/error_prone_core So I doubt it is a version problem.

Apparently it never ran on the modules which are now failing.

**Comment by kennknowles** [comment]
Should move the discussion of how to unbreak the build to dev@. This PR is just a collection of attempted fixes, which I'll probably continue to power through.

**Comment by swegner** [comment]
Some probably-unrelated context: we're using a fairly old version of the errorprone gradle plugin; I [made an attempt to upgrade](https://issues.apache.org/jira/browse/BEAM-5400) about a month ago and hit a number of issues along the way.

I believe the gradle-errorprone-plugin version is decoupled from the actual ErrorProne tool version. I know we have the gradle plugin version pinned, but I wonder if we're not pinning the ErrorProne tool version. I wonder if IntelliJ is picking a different version for you.

**Comment by kennknowles** [comment]
The version of the plugin is decoupled from the version of errorprone. The plugin's docs strongly caution you to pin a version so your build doesn't break arbitrarily.

**Comment by kennknowles** [comment]
Since errorprone is now pinned, no need for this until it is time to bump it up.

**Pull request closed without merging**